pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -564,14 +565,21 @@ async def on_complete():
                                         messages.append(_messages.ModelRequest(parts))
                                     run_span.set_attribute('all_messages', messages)
 
+                                # The following is not guaranteed to be true, but we consider it a user error if
```

Comment:
Is this something that should be added to the docstrings or project docs so that users are aware of this scenario?

Reply:
Yes, good suggestion